contracts/test/strategies/oeth-metapool.fork-test.js

```diff
@@ -1,309 +1,863 @@
+      await expect(tx)
+        .to.emit(convexEthMetaStrategy, "Withdrawal")
+        .withArgs(weth.address, oethMetaPool.address, withdrawAmount);
+      await expect(tx)
+        .to.emit(convexEthMetaStrategy, "Withdrawal")
+        .withNamedArgs({ _asset: oeth.address, _pToken: oethMetaPool.address });
+
+      // Check the ETH and OETH balances in the Curve Metapool
+      const curveBalancesAfter = await oethMetaPool.get_balances();
+      expect(curveBalancesAfter[0]).to.approxEqualTolerance(
+        curveBalancesBefore[0].sub(withdrawAmount),
+        0.01 // 0.01% or 1 basis point
+      );
+      expect(curveBalancesAfter[1]).to.approxEqualTolerance(
+        curveBalancesBefore[1].sub(oethBurnAmount),
+        0.01 // 0.01%
+      );
 
-  it("Should be able to withdraw all", async () => {
-    const { oethVault, oeth, weth, josh, ConvexEthMetaStrategy } =
-      await loadFixture(convexOETHMetaVaultFixture);
+      // Check the OETH total supply decrease
+      const oethSupplyAfter = await oeth.totalSupply();
+      expect(oethSupplyAfter).to.approxEqualTolerance(
+        oethSupplyBefore.sub(oethBurnAmount),
+        0.01 // 0.01% or 1 basis point
+      );
 
-    await oethVault.connect(josh).allocate();
-    const supplyBeforeMint = await oeth.totalSupply();
-    const amount = "10";
-    const unitAmount = oethUnits(amount);
+      // Check the WETH balance in the Vault
+      expect(await weth.balanceOf(oethVault.address)).to.equal(
+        vaultWethBalanceBefore.add(withdrawAmount)
+      );
+    });
+    it("Only vault can withdraw some WETH from AMO strategy", async function () {
+      const {
+        convexEthMetaStrategy,
+        oethVault,
+        strategist,
+        timelock,
+        josh,
+        weth,
+      } = fixture;
+
+      for (const signer of [strategist, timelock, josh]) {
+        const tx = convexEthMetaStrategy
+          .connect(signer)
+          .withdraw(oethVault.address, weth.address, parseUnits("50"));
+
+        await expect(tx).to.revertedWith("Caller is not the Vault");
+      }
+    });
+    it("Only vault and governor can withdraw all WETH from AMO strategy", async function () {
+      const { convexEthMetaStrategy, strategist, timelock, josh } = fixture;
+
+      for (const signer of [strategist, josh]) {
+        const tx = convexEthMetaStrategy.connect(signer).withdrawAll();
+
+        await expect(tx).to.revertedWith("Caller is not the Vault or Governor");
+      }
+
+      // Governor can withdraw all
+      const tx = convexEthMetaStrategy.connect(timelock).withdrawAll();
+      await expect(tx).to.emit(convexEthMetaStrategy, "Withdrawal");
+    });
+  });
 
-    await weth.connect(josh).approve(oethVault.address, unitAmount);
-    await oethVault.connect(josh).mint(weth.address, unitAmount, 0);
-    await oethVault.connect(josh).allocate();
+  describe("with a lot more OETH in the Metapool", () => {
+    const loadFixture = createFixtureLoader(convexOETHMetaVaultFixture, {
```

Comment:
I think these might fail if pool on the mainnet get heavily out of balance. Or at least they won't end up with the required tilt

Reply:
The tx won't fail if the tilt is improved.